src/ray/raylet/node_manager.cc

```diff
@@ -1284,6 +1316,24 @@ void NodeManager::ProcessNotifyActorResumedFromCheckpoint(const uint8_t *message
   checkpoint_id_to_restore_.emplace(actor_id, checkpoint_id);
 }
 
+void NodeManager::ProcessActiveObjectIDsHeartbeat(
+    const std::shared_ptr<LocalClientConnection> &client, const uint8_t *message_data) {
+  std::shared_ptr<Worker> worker = worker_pool_.GetRegisteredWorker(client);
+  if (!worker) {
+    worker = worker_pool_.GetRegisteredDriver(client);
+    RAY_CHECK(worker);
+  }
+
+  auto message = flatbuffers::GetRoot<protocol::ActiveObjectIDsHeartbeat>(message_data);
+  std::vector<ObjectID> object_ids = from_flatbuf<ObjectID>(*message->object_ids());
+
+  std::unordered_set<ObjectID> active_object_ids;
```

Comment:
I think you can initialize with `std::unordered_set<ObjectID> active_object_ids(object_ids.begin(), object_ids.end());`. Might be faster.